server/src/main/java/io/crate/analyze/AlterTableDropColumnAnalyzer.java

```diff
@@ -52,7 +66,70 @@ public AnalyzedAlterTableDropColumn analyze(AlterTableDropColumn<Expression> alt
             txnCtx.sessionSettings().sessionUser(),
             txnCtx.sessionSettings().searchPath());
 
-        var analyzer = new TableElementsAnalyzer(tableInfo, txnCtx, nodeCtx, paramTypeHints);
-        return analyzer.analyze(alterTable);
+        var expressionAnalyzer = new ExpressionAnalyzer(
+            txnCtx,
+            nodeCtx,
+            paramTypeHints,
+            new NameFieldProvider(new DocTableRelation(tableInfo)),
+            null
+        );
+        var expressionContext = new ExpressionAnalysisContext(txnCtx.sessionSettings());
+        List<Reference> dropColumns = new ArrayList<>(alterTable.tableElements().size());
+
+        for (var dropColumnDefinition : alterTable.tableElements()) {
+            Expression name = dropColumnDefinition.name();
+            try {
+                Symbol columnSymbol = expressionAnalyzer.convert(name, expressionContext);
+                ColumnIdent columnName = Symbols.pathFromSymbol(columnSymbol);
+                Reference reference = tableInfo.getReference(columnName);
```

Comment:
Isn't the `columnSymbol` already a reference here?
Does the parser allow expressions other then idents? If so, maybe make sure we throw a sensible error if a function or literal is used

Reply:
Yep it is, I added simply a cast.
yep, parser doesn't allow fn calls or literals.